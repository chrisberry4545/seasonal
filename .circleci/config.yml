version: 2.1

commands:
  install-dependencies:
    steps:
      - restore_cache:
          name: Restore dependency cache
          keys:
            - seasonal-deps-{{ checksum "package-lock.json" }}
      - run:
          name: Install dependencies
          command: |
            npm i
            npm run bootstrap
      - save_cache:
          name: Save dependency cache
          paths:
            - node_modules
          key: seasonal-deps-{{ checksum "package-lock.json" }}

jobs:
  backend_test:
    docker:
      - image: circleci/node:10.15
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: Run backend tests
          command: ./tests/backend-run-functional-tests.sh

  backend_deploy:
    docker:
      - image: circleci/node:10.15
    steps:
      - checkout
      - install-dependencies
      - run:
          name: Copy env config
          command: |
            cd ./packages/backend
            cp .env-ci.yml .env.yml
      - run:
          name: Build dist version
          command: |
            cd ./packages/backend
            npm run build
            npm run move-dist-to-top-level
      - run:
          name: Install Serverless CLI
          command: sudo npm i -g serverless
      - run:
          name: Deploy backend
          command: |
            cd ./packages/backend
            sls deploy --stage prod

  frontend_deploy:
    docker:
      - image: circleci/node:10.15
    steps:
      - checkout
      - install-dependencies
      - add_ssh_keys:
          fingerprints:
            - cd:42:e8:de:36:e3:c0:f9:0d:6f:55:27:47:57:ab:4b
      - run:
          name: Git configuration
          command: |
            git config user.email "chrisbery4545@googlemail.com"
            git config user.name "chrisbery4545"
      - run:
          name: Deploy frontend
          command: |
            cd ./packages/frontend
            npm run deploy

workflows:
  version: 2
  backend:
    jobs:
      - backend_test
      - backend_approval:
          type: approval
          filters:
            branches:
              only:
                - master
          requires:
           - backend_test
      - backend_deploy:
          requires:
            - backend_approval
      - frontend_approval:
          type: approval
          filters:
            branches:
              only:
                - master
      - frontend_deploy:
          requires:
            - frontend_approval
