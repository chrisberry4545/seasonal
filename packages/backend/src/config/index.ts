export * from './env-vars';

export * from './urls';
