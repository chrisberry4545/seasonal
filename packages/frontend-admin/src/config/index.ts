const BACKEND_URL = process.env.BACKEND_URL
  || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/admin';

export const LOGIN_URL =
  process.env.LOGIN_URL || `${BACKEND_URL}/login`;

export const RECIPES_URL =
  process.env.RECIPES_URL || `${BACKEND_URL}/recipes`;
