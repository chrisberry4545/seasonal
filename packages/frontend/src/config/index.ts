const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
    || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/v1';

export const SEASON_DATA_URL =
  process.env.REACT_APP_SEASON_DATA_URL || `${BACKEND_URL}/season-data`;

export const FOOD_DETAILS_DATA_URL =
  process.env.REACT_APP_FOOD_DETAILS_DATA_URL || `${BACKEND_URL}/food-data`;
