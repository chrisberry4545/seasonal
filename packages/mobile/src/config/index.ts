import { Platform } from 'react-native';

export const PRIVACY_POLICY_URL: string =
  'https://chrisberry4545.github.io/eat-seasonal-privacy-policy/';
export const STORE_URL: string = Platform.OS === 'ios'
  ? 'https://play.google.com/store/apps/details?id=com.chrisbdev.seasonal'
  : 'market://details?id=com.chrisbdev.seasonal';
