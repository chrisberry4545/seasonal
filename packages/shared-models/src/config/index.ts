const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
  || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/v2';

export const SEASON_URL =
  process.env.SEASON_URL || `${BACKEND_URL}/season`;

export const FOOD_DETAILS_URL =
  process.env.FOOD_DETAILS_URL || `${BACKEND_URL}/food`;

export const SEASON_WITH_FOOD_URL =
  process.env.SEASON_WITH_FOOD_URL || `${BACKEND_URL}/season-with-food`;

export const SEASON_WITH_RECIPES_URL =
  process.env.SEASON_WITH_RECIPES_URL || `${BACKEND_URL}/season-with-recipes`;

export const COUNTRY_URL =
  process.env.COUNTRY_URL || `${BACKEND_URL}/country`;
