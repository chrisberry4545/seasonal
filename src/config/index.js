const BACKEND_URL = process.env.BACKEND_URL
    || 'http://localhost:3000';

const config = {
    SEASON_DATA_URL: process.env.SEASON_DATA_URL || `${BACKEND_URL}/season-data`
};

export {
    config
};
