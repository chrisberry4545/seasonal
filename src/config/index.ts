const BACKEND_URL: string = process.env.BACKEND_URL
  || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/v1';

export const SEASON_DATA_URL: string =
  process.env.SEASON_DATA_URL || `${BACKEND_URL}/season-data`;

export const FOOD_DETAILS_DATA_URL =
  process.env.FOOD_DETAILS_DATA_URL || `${BACKEND_URL}/food-data`;

export const PRIVACY_POLICY_URL: string =
  'https://chrisberry4545.github.io/eat-seasonal-privacy-policy/';
