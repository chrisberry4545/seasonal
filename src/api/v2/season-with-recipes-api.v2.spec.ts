import {
  airtableMockSetup
} from '../mocks';
airtableMockSetup();

import {
  app
} from '../../app';

import supertest, { Response } from 'supertest';
import { V2_ENDPOINT, SEASON_WITH_RECIPES_ENDPOINT } from '../../config';

const v2SeasonWithRecipesUrl = `${V2_ENDPOINT}/${SEASON_WITH_RECIPES_ENDPOINT}`;

describe('Get all seasons with recipes', () => {
  let response: Response;
  beforeEach(async () => {
    response = await supertest(app).get(`/${v2SeasonWithRecipesUrl}`);
  });

  test('Returns a status of 200', () => {
    expect(response.status).toBe(200);
  });
  test('Returns a full list of season data', () => {
    expect(response.body).toMatchSnapshot();
  });
  test('Returns the name of the first season', () => {
    expect(response.body[0].name).toBe('January');
  });
  test('Returns the name of the second season', () => {
    expect(response.body[1].name).toBe('February');
  });
  test('Populates the recipes in the seasons', () => {
    expect(response.body[0].recipes).toHaveLength(1);
  });
  test('Sets the recipes in the seasons name', () => {
    expect(response.body[0].recipes[0].name).toBe('recipe 1');
  });
});

const makeSingleSeasonWithRecipesRequest = (id: string = '0') => {
  return supertest(app).get(`/${v2SeasonWithRecipesUrl}/${id}`);
};

describe('Get single season with recipes', () => {
  let response: Response;

  describe('when the season has no recipes', () => {
    beforeAll(async () => {
      response = await makeSingleSeasonWithRecipesRequest();
    });

    test('Returns a status of 200', () => {
      expect(response.status).toBe(200);
    });
    test('Retrieves a single season', () => {
      expect(response.body).toMatchSnapshot();
    });
    test('Does not populate a seasons recipes if they do not exist', () => {
      expect(response.body.recipes).toHaveLength(0);
    });
  });

  describe('when the season has recipes', () => {
    beforeAll(async () => {
      response = await makeSingleSeasonWithRecipesRequest('1');
    });

    test('Retrieves a single season with food and recipe data', () => {
      expect(response.body).toMatchSnapshot();
    });
    test('Populates a seasons recipes if they exist', () => {
      expect(response.body.recipes.length > 0).toBe(true);
    });
    test('Populates a seasons recipes isVegetarian if true', () => {
      expect(response.body.recipes[0].isVegetarian).toBe(true);
    });
    test('Populates a seasons recipes isVegan if true', () => {
      expect(response.body.recipes[1].isVegan).toBe(true);
    });
    test('Does not return any primaryFood on a recipe', () => {
      expect(response.body.recipes[0].primaryFood).toBeUndefined();
    });
    test('Does not return any secondaryFood on a recipe', () => {
      expect(response.body.recipes[0].secondaryFood).toBeUndefined();
    });
    test('Does not return any food', () => {
      expect(response.body.food).toBeUndefined();
    });
  });
});
