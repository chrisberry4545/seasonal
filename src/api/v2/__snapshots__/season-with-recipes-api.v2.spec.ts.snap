// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get all seasons with recipes Returns a full list of season data 1`] = `
Array [
  Object {
    "id": "0",
    "name": "January",
    "recipes": Array [
      Object {
        "id": "r1",
        "isVegan": true,
        "name": "recipe 1",
      },
    ],
  },
  Object {
    "id": "1",
    "name": "February",
    "recipes": Array [
      Object {
        "id": "r1",
        "isVegan": true,
        "name": "recipe 1",
      },
    ],
  },
]
`;

exports[`Get single season with recipes when an isVegan filter is applied Retrieves the expected data 1`] = `
Object {
  "id": "1",
  "name": "February",
  "recipes": Array [
    Object {
      "id": "r1",
      "isVegan": true,
      "name": "recipe 1",
    },
  ],
}
`;

exports[`Get single season with recipes when an isVegetarian filter is applied Retrieves the expected data 1`] = `
Object {
  "id": "1",
  "name": "February",
  "recipes": Array [
    Object {
      "id": "r2",
      "isVegetarian": true,
      "name": "other recipe",
    },
    Object {
      "id": "r1",
      "isVegan": true,
      "name": "recipe 1",
    },
  ],
}
`;

exports[`Get single season with recipes when both isVegan and isVegetarian filter is applied Retrieves the expected data 1`] = `
Object {
  "id": "1",
  "name": "February",
  "recipes": Array [
    Object {
      "id": "r1",
      "isVegan": true,
      "name": "recipe 1",
    },
  ],
}
`;

exports[`Get single season with recipes when the season has no recipes Retrieves a single season 1`] = `
Object {
  "id": "0",
  "name": "January",
  "recipes": Array [],
}
`;

exports[`Get single season with recipes when the season has recipes Retrieves a single season with food and recipe data 1`] = `
Object {
  "id": "1",
  "name": "February",
  "recipes": Array [
    Object {
      "id": "r2",
      "isVegetarian": true,
      "name": "other recipe",
    },
    Object {
      "id": "r3",
      "name": "other recipe 2",
    },
    Object {
      "id": "r1",
      "isVegan": true,
      "name": "recipe 1",
    },
  ],
}
`;
